From 6f76e3a7c922b4b5ed650ab406d5f6f42a51f31b Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 17 Sep 2024 10:47:15 +0200
Subject: [PATCH] drm/stm: lvds: fix support of 2 singles data channels

Need to check link type "single" & "secondary" to avoid noisy message
"No phase precised, setting default".

Change-Id: I3d47b3aa9b975de1a983b75b27e0e9a260a0cb98
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/404868
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/437322
---
 drivers/gpu/drm/stm/lvds.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/gpu/drm/stm/lvds.c
+++ b/drivers/gpu/drm/stm/lvds.c
@@ -862,6 +862,8 @@ static void lvds_config_mode(struct stm_
 		lvds_cr |= CR_VSPOL;
 
 	switch (lvds->link_type) {
+	case LVDS_SINGLE_LINK_PRIMARY:
+	case LVDS_SINGLE_LINK_SECONDARY:
 	case LVDS_DUAL_LINK_EVEN_ODD_PIXELS: /* LKPHA = 0 */
 		lvds_cr &= ~CR_LKPHA;
 		break;
